app/src/main/java/de/danoeh/antennapod/fragment/EpisodesListFragment.java

```diff
@@ -441,14 +446,23 @@ void loadItems() {
                         });
     }
 
+    private SortOrder getSortOrder() {
+        SharedPreferences prefs = getActivity().getSharedPreferences(AllEpisodesFragment.PREF_NAME,
+                Context.MODE_PRIVATE);
+        String sortOrderStr = prefs.getString(PREF_SORT, "DATE_NEW_OLD");
+        return SortOrder.parseWithDefault(sortOrderStr, SortOrder.DATE_NEW_OLD);
+    }
+
     @NonNull
     protected List<FeedItem> loadData() {
-        return DBReader.getRecentlyPublishedEpisodes(0, page * EPISODES_PER_PAGE, getFilter());
+        return DBReader.getRecentlyPublishedEpisodes(0, page * EPISODES_PER_PAGE,
+                getFilter(), getSortOrder());
     }
 
     @NonNull
     protected List<FeedItem> loadMoreData(int page) {
-        return DBReader.getRecentlyPublishedEpisodes((page - 1) * EPISODES_PER_PAGE, EPISODES_PER_PAGE, getFilter());
+        return DBReader.getRecentlyPublishedEpisodes((page - 1) * EPISODES_PER_PAGE,
+                EPISODES_PER_PAGE, getFilter(), getSortOrder());
```

Comment:
Please simply overwrite the `loadData` and `loadMoreData` (and `loadTotalItemCount`) methods in `AllEpisodesFragment`. Then you don't need to access the preferences object of the child class. Also, while we are at it, you could make the methods here `abstract`.